### ISSUE TITLE
Rule proposal: forbid imports

### ISSUE THREAD

**Issue by lencioni** [benmosher/eslint-plugin-import#501]
We might have a package that we want to use in one very specific place, but prevent it from being used elsewhere. It would be nice to have a rule that could be customized to prevent it from being imported anywhere. 

It could take an object of package name keys and message pairs to give high-quality messages.

Options:
```json
"import/forbid-imports": [2, {
  "some-package": "Please use some-other-package instead.",
}],
```

```js
// Bad
import foo from 'some-package';
import { bar } from 'some-package';
const foo = require('some-package');
const { bar } = require('some-package');

// Good
import foo from 'some-other-package';
```

Thoughts?

**Comment by lencioni** [comment]
I suppose if we can make #500 work, that would probably satisfy this scenario pretty well.

**Comment by ljharb** [comment]
I think the core rule [no-restricted-imports](http://eslint.org/docs/rules/no-restricted-imports) may cover this already.

**Comment by benmosher** [comment]
@ljharb seems to be right. Might be a good enhancement for the core rule.

**Comment by lencioni** [comment]
Indeed. I realize I wasn't very clear, but that's what I was trying to say about making #500 work. :)

**Comment by lencioni** [comment]
Ah, you are right! This is already a rule. Thanks! 👍
